internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -773,35 +775,66 @@ func expandLabelSliceFromSetSchema(d *schema.ResourceData) ([]admin.ComponentLab
 	return res, nil
 }
 
-func expandAdvancedReplicationSpecs(tfList []any) *[]admin.ReplicationSpec {
+func expandAdvancedReplicationSpecs(tfList []any, rootDiskSizeGB *float64) *[]admin.ReplicationSpec20240710 {
 	if len(tfList) == 0 {
 		return nil
 	}
-	var apiObjects []admin.ReplicationSpec
+	var apiObjects []admin.ReplicationSpec20240710
 	for _, tfMapRaw := range tfList {
 		tfMap, ok := tfMapRaw.(map[string]any)
 		if !ok || tfMap == nil {
 			continue
 		}
-		apiObject := expandAdvancedReplicationSpec(tfMap)
+		apiObject := expandAdvancedReplicationSpec(tfMap, rootDiskSizeGB)
+		apiObjects = append(apiObjects, *apiObject)
+
+		// handles adding additional replication spec objects if legacy num_shards attribute is being used and greater than 1
+		numShards := tfMap["num_shards"].(int)
+		for range numShards - 1 {
+			apiObjects = append(apiObjects, *apiObject)
+		}
+	}
+	return &apiObjects
+}
+
+func expandAdvancedReplicationSpecsOldSDK(tfList []any) *[]admin20231115.ReplicationSpec {
+	if len(tfList) == 0 {
+		return nil
+	}
+	var apiObjects []admin20231115.ReplicationSpec
+	for _, tfMapRaw := range tfList {
+		tfMap, ok := tfMapRaw.(map[string]any)
+		if !ok || tfMap == nil {
+			continue
+		}
+		apiObject := expandAdvancedReplicationSpecOldSDK(tfMap)
 		apiObjects = append(apiObjects, *apiObject)
 	}
 	return &apiObjects
 }
 
-func expandAdvancedReplicationSpec(tfMap map[string]any) *admin.ReplicationSpec {
-	apiObject := &admin.ReplicationSpec{
+func expandAdvancedReplicationSpec(tfMap map[string]any, rootDiskSizeGB *float64) *admin.ReplicationSpec20240710 {
+	apiObject := &admin.ReplicationSpec20240710{
+		ZoneName:      conversion.StringPtr(tfMap["zone_name"].(string)),
+		RegionConfigs: expandRegionConfigs(tfMap["region_configs"].([]any), rootDiskSizeGB),
+	}
+	// here we will populate id value using external_id value (relevant for update)
```

Comment:
nit: the comments are a bit confusing, can we mark relevant ones as `TODO`s for now so it's easy to keep track

Reply:
makes sense, adjusted